packages/uploadthing/test/__test-helpers.ts

```diff
@@ -94,111 +77,233 @@ const mockPresigned = (file: {
+
+        utApiMock(request.url, {
+          method: request.method,
+          headers: Object.fromEntries(request.headers.entries()),
+          body: JSON.stringify(body),
+        });
+
+        const presigneds = body.files.map((file) => {
           const presigned = mockPresigned(file);
           db.insertFile({
             ...file,
             key: presigned.key,
+            callbackUrl: body.callbackUrl,
+            callbackSlug: body.callbackSlug,
           });
           return presigned;
         });
-        return Response.json({ data: presigneds });
-      }
-      if (url.pathname === "/api/prepareUpload") {
-        const presigneds = (body?.files as any[]).map((file) => {
+        return HttpResponse.json(presigneds);
+      },
+    ),
+    http.post(
+      "https://uploadthing.com/api/uploadFiles",
+      async ({ request }) => {
+        const body = (await request.json()) as { files: any[] };
+
+        utApiMock(request.url, {
+          method: request.method,
+          headers: Object.fromEntries(request.headers.entries()),
+          body: JSON.stringify(body),
+        });
+
+        const presigneds = body?.files.map((file) => {
           const presigned = mockPresigned(file);
           db.insertFile({
             ...file,
             key: presigned.key,
-            callbackUrl: body!.callbackUrl,
-            callbackSlug: body!.callbackSlug,
           });
           return presigned;
         });
-        return Response.json(presigneds);
-      }
-      if (url.pathname === "/api/completeMultipart") {
-        return Response.json({ success: true });
-      }
-      if (url.pathname === "/api/failureCallback") {
-        return Response.json({ success: true });
-      }
-      if (url.pathname.startsWith("/api/pollUpload")) {
+        return HttpResponse.json({ data: presigneds });
+      },
+    ),
+    http.post(
+      "https://uploadthing.com/api/completeMultipart",
+      async ({ request }) => {
+        const body = await request.json();
+
+        utApiMock(request.url, {
+          method: request.method,
+          headers: Object.fromEntries(request.headers.entries()),
+          body: JSON.stringify(body),
+        });
+
+        return HttpResponse.json({ success: true });
+      },
+    ),
+    http.post(
+      "https://uploadthing.com/api/failureCallback",
+      async ({ request }) => {
+        const body = await request.json();
+
+        utApiMock(request.url, {
+          method: request.method,
+          headers: Object.fromEntries(request.headers.entries()),
+          body: JSON.stringify(body),
+        });
+
+        return HttpResponse.json({ success: true });
+      },
+    ),
+    http.get(
+      "https://uploadthing.com/api/pollUpload/:key",
+      async ({ request }) => {
+        utApiMock(request.url, {
+          method: request.method,
+          headers: Object.fromEntries(request.headers.entries()),
+        });
+
+        const url = new URL(request.url);
```

Comment:
Looks like you aren't doing anything with this `url`. A left-over code?

Reply:
I am reading the key from the dynamic url. But I think I can use `params` for that instead right?

Reply:
Yes. MSW already parses the request path parameters for you.

```ts
http.get('/:key', ({ params }) => params.key)
```

You can also provide type annotations for parameters:

```ts
http.get<{ key: string }>('/:key', resolver)
```